ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +56,28 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
```

Comment:
Prior to your fix would this new test have produced the exception reported in the JIRA:

> When tx abort, and we have asyncAfterSynchronization option, there's exception

If it did not then would like to see a test that acts a reproducer.